src/CommunityToolkit.Maui/Extensions/Popup/PopupExtensions.shared.cs

```diff
@@ -0,0 +1,108 @@
+﻿using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Views;
+using Microsoft.Maui.Platform;
+
+namespace Microsoft.Maui.Controls;
+
+/// <summary>
+/// Extension methods for <see cref="Popup"/>.
+/// </summary>
+public static partial class PopupExtensions
+{
+	/// <summary>
+	/// Displays a popup.
+	/// </summary>
+	/// <param name="page">
+	/// The current <see cref="Page"/>.
+	/// </param>
+	/// <param name="popup">
+	/// The <see cref="BasePopup"/> to display.
+	/// </param>
+	public static void ShowPopup<TPopup>(this Page page, TPopup popup) where TPopup : BasePopup
+		=> ShowPopup(page.Navigation, popup);
+
+	/// <summary>
+	/// Displays a popup.
+	/// </summary>
+	/// <param name="navigation">
+	/// The current <see cref="INavigation"/>.
+	/// </param>
+	/// <param name="popup">
+	/// The <see cref="BasePopup"/> to display.
+	/// </param>
+	public static void ShowPopup<TPopup>(this INavigation navigation, TPopup popup) where TPopup : BasePopup
+	{
+#if WINDOWS
+		PlatformShowPopup(popup, GetMauiContext(navigation));
+#else
+		CreatePopup(navigation, popup);
+#endif
+
+	}
+
+	/// <summary>
+	/// Displays a popup and returns a result.
+	/// </summary>
+	/// <param name="page">
+	/// The current <see cref="Page"/>.
+	/// </param>
+	/// <param name="popup">
+	/// The <see cref="Popup"/> to display.
+	/// </param>
+	/// <returns>
+	/// A task that will complete once the <see cref="Popup"/> is dismissed.
+	/// </returns>
+	public static Task<object?> ShowPopupAsync<TPopup>(this Page page, TPopup popup) where TPopup : Popup
```

Comment:
@brminnick 

We have a situation here, where we need to decide if we will use `Page` or `INavigation` (to make things easier to us). I just tested the code and if we do this:
```csharp
var page = new ContentPage
{
	Content = new Label
	{
		Text = "Hello there"
	}
};

page.ShowPopup((MockPopup)popup);
```

this will fail, because the `page.Navigation.NavigationStack.Count` will be zero and this will throw an exception on the `GetMauiContext` method (because we will not have anything to grab the `Handler.MauiContext`). Now if we rewrite this API to work with `Page` we can grab the `MauiContext`. So my vote here is to make it work with `Page`.

The code that works (using INavigation) is
```csharp
		var page = new ContentPage
		{
			Content = new Label
			{
				Text = "Hello there"
			}
		};
		var np = new NavigationPage(page);
		page.ShowPopup((MockPopup)popup);
```
Sadly this isn't obvious for devs that not creating a `NavigationPage` will throw an exception. thoughts??

Reply:
Agreed that we should use `Page`.

Seeing that you're referencing `MockPopup`, I assume that you're getting that exception while unit testing. If that's the case, you'll need to create a `ViewHandler` that implements its own `MauiContext` like we did for MockAnimationHandler:

https://github.com/CommunityToolkit/Maui/blob/c4d7d72b1512a187f9ca0d1560e00dcab3fe4543/src/CommunityToolkit.Maui.UnitTests/Mocks/AnimationExtensions.cs#L11-L141

Reply:
Thanks for the reference

Reply:
First I'll update the API to use `Page` and then I move to tests